src/Compilers/CSharp/Portable/Binder/ExpressionVariableBinder.cs

```diff
@@ -21,7 +21,8 @@ internal ExpressionVariableBinder(SyntaxNode scopeDesignator, Binder next) : bas
         protected override ImmutableArray<LocalSymbol> BuildLocals()
         {
             var builder = ArrayBuilder<LocalSymbol>.GetInstance();
-            ExpressionVariableFinder.FindExpressionVariables(this, builder, (CSharpSyntaxNode)ScopeDesignator);
+            ExpressionVariableFinder.FindExpressionVariables(this, builder, (CSharpSyntaxNode)ScopeDesignator, 
+                                                             GetBinder((CSharpSyntaxNode)ScopeDesignator));
```

Comment:
Yes, I think it would be better, though I would prefer to make this API change separately (https://github.com/dotnet/roslyn/issues/13196).